gammapy/cube/tests/test_simulate.py

```diff
@@ -0,0 +1,52 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import numpy as np
+from astropy.coordinates import SkyCoord
+import astropy.units as u
+from ...irf import load_cta_irfs
+from ...maps import WcsGeom, MapAxis
+from ...spectrum.models import PowerLaw
+from ...image.models import SkyGaussian
+from ...cube.models import SkyModel
+from ...cube import MapDataset
+from ..simulate import simulate_3d
+
+
+def test_simulate():
+    filename = (
+        "$GAMMAPY_DATA/cta-1dc/caldb/data/cta/1dc/bcf/South_z20_50h/irf_file.fits"
+    )
+    irfs = load_cta_irfs(filename)
+
+    # Define sky model to simulate the data
+    spatial_model = SkyGaussian(lon_0="0 deg", lat_0="0 deg", sigma="0.2 deg")
+    spectral_model = PowerLaw(
+        index=2, amplitude="1e-11 cm-2 s-1 TeV-1", reference="1 TeV"
+    )
+    sky_model_simu = SkyModel(
+        spatial_model=spatial_model, spectral_model=spectral_model
+    )
+
+    # Define map geometry
+    axis = MapAxis.from_edges(
+        np.logspace(-1, 1.0, 10), unit="TeV", name="energy", interp="log"
+    )
+    geom = WcsGeom.create(
+        skydir=(0, 0), binsz=0.05, width=(1, 1), coordsys="GAL", axes=[axis]
+    )
+
+    # Define some observation parameters
+    pointing = SkyCoord(0 * u.deg, 0 * u.deg, frame="galactic")
+
+    dataset = simulate_3d(
+        sky_model_simu, geom, pointing, irfs, livetime=1 * u.h, edisp=True
+    )
+
+    assert isinstance(dataset, MapDataset)
+    assert isinstance(dataset.model, SkyModel)
+
+    assert dataset.counts.data.dtype is np.dtype("int")
+    assert np.sum(dataset.counts.data) > 0
```

Comment:
Can you use `assert_allclose` statements here and assert one the actual values returned by this test? If the assert is only > 0, we'll never notice, when something changed in the code...

Reply:
I'm not sure how to do this, so you would want to test that sum(counts) returns always the same value ? I guess we need to fix the seed for that (but this is doable now).
In addition if there is a little change in the way the IRF  integration is done this will change the Npred and the counts, the exposure, the bkg, and the edisp will also be affected.
Is that ok ?
My first check was to make sure that we don't return empty arrays.

Reply:
Yes, that's definitely OK. I think test should be as sensitive as possible to any change in the code. I guess the best option here is to test on a single pixel:

```python3
from numpy.testing import assert_allclose

assert_allclose(dataset.counts.data[2, 24, 24], 3)
assert_allclose(dataset.exposure.data[2, 24, 24], 1e10)
etc.
```
You can just copy and paste the reference values from the test dataset. Tests should be reproducible as well, so it should be called with a random seed (however I think `random_state='random-seed'` is already a seed, so probably no change is required here).

Reply:
ok, just modified the tests to assert_allclose()